File: app/src/main/java/io/github/grebenindmitry/nutrado/Converters.java
package io.github.grebenindmitry.nutrado;

import android.graphics.Color;
import androidx.room.TypeConverter;

public class Converters {
    @TypeConverter
    public float nutrimentsToFloat(ProductNutriments productNutriments) {
        return productNutriments.getEnergy();
    }

    @TypeConverter
    public ProductNutriments floatToNutriments(float energy) {
        return new ProductNutriments(energy);
    }

    @TypeConverter
    public Color intToColor(int color) {
        return Color.valueOf(color);
    }

    @TypeConverter
    public int colorToInt(Color color) {
        return color.toArgb();
    }
}

File: app/src/main/java/io/github/grebenindmitry/nutrado/Converters.kt
package io.github.grebenindmitry.nutrado

import android.graphics.Color
import androidx.room.TypeConverter

class Converters {
    @TypeConverter
    fun nutrimentsToFloat(productNutriments: StructNutriments): Float { return productNutriments.energy }

    @TypeConverter
    fun floatToNutriments(energy: Float): StructNutriments { return StructNutriments(energy) }

    @TypeConverter
    fun intToColor(color: Int): Color { return Color.valueOf(color) }

    @TypeConverter
    fun colorToInt(color: Color): Int { return color.toArgb() }
}